zerver/views/realm_export.py

```diff
@@ -37,10 +37,13 @@ def export_realm(request: HttpRequest, user: UserProfile) -> HttpResponse:
     # Filter based upon the number of events that have occurred in the delta
     # If we are at the limit, the incoming request is rejected
     event_time_delta = event_time - timedelta(days=7)
-    limit_check = RealmAuditLog.objects.filter(
-        realm=realm, event_type=event_type, event_time__gte=event_time_delta
+    successful_recent_export_count = RealmAuditLog.objects.filter(
+        realm=realm,
+        event_type=event_type,
+        event_time__gte=event_time_delta,
+        extra_data__failed_timestamp__isnull=True,
     ).count()
-    if limit_check >= EXPORT_LIMIT:
+    if successful_recent_export_count >= EXPORT_LIMIT:
         raise JsonableError(_("Exceeded rate limit."))
```

Comment:
I think the original design intent was to count failures, because they are likely to be correlated with exports that place undue load on our infrastructure, and the rate limit is an anti-DoS measure.

Reply:
Hm, OK.  We had users who were running into server-side errors which were not their fault or control; once we resolved them, they were unable to export because of the number of failures they'd had.

In general, however, most folks will not be able to _intentionally_ cause a failed export, so it's difficult to abuse.

In this case, I did a manual export for them, which is I guess an OK solution in general?

The first commit should still be merged, however.

Reply:
Yeah, I think doing a manual export is a fine answer there's; it's hopefully not much overhead given that we probably needed to fix the actual server-side root cause anyway.

Reply:
[#29697](https://github.com/zulip/zulip/pull/29697) is the fix.